src/EditorFeatures/Core/Implementation/AutomaticCompletion/AbstractAutomaticLineEnderCommandHandler.cs

```diff
@@ -84,54 +81,53 @@ public void ExecuteCommand(AutomaticLineEnderCommandArgs args, Action nextHandle
+                if (TreatAsReturn(document, position.Value.Position, w.CancellationToken))
                 {
-                    // caret is not on the subject buffer. nothing we can do
-                    var position = args.TextView.GetCaretPoint(args.SubjectBuffer);
-                    if (!position.HasValue)
-                    {
-                        NextAction(operations, nextHandler);
-                        return;
-                    }
-
-                    var subjectLineWhereCaretIsOn = position.Value.GetContainingLine();
-                    var insertionPoint = GetInsertionPoint(document, subjectLineWhereCaretIsOn, w.CancellationToken);
-                    if (!insertionPoint.HasValue)
-                    {
-                        NextAction(operations, nextHandler);
-                        return;
-                    }
-
-                    // special cases where we treat this command simply as Return.
-                    if (TreatAsReturn(document, position.Value.Position, w.CancellationToken))
-                    {
-                        // leave it to the VS editor to handle this command.
-                        // VS editor's default implementation of SmartBreakLine is simply BreakLine, which inserts
-                        // a new line and positions the caret with smart indent.
-                        nextHandler();
-                        return;
-                    }
-
-                    using (var transaction = args.TextView.CreateEditTransaction(EditorFeaturesResources.Automatic_Line_Ender, _undoRegistry, _editorOperationsFactoryService))
-                    {
-                        // try to move the caret to the end of the line on which the caret is
-                        args.TextView.TryMoveCaretToAndEnsureVisible(subjectLineWhereCaretIsOn.End);
-
-                        // okay, now insert ending if we need to
-                        var newDocument = InsertEndingIfRequired(document, insertionPoint.Value, position.Value, w.CancellationToken);
-
-                        // format the document and apply the changes to the workspace
-                        FormatAndApply(newDocument, insertionPoint.Value, w.CancellationToken);
-
-                        // now, insert new line
-                        NextAction(operations, nextHandler);
-
-                        transaction.Complete();
-                    }
-                });
+                    // leave it to the VS editor to handle this command.
+                    // VS editor's default implementation of SmartBreakLine is simply BreakLine, which inserts
+                    // a new line and positions the caret with smart indent.
+                    nextHandler();
+                    return;
+                }
+
+                using (var transaction = args.TextView.CreateEditTransaction(EditorFeaturesResources.Automatic_Line_Ender, _undoRegistry, _editorOperationsFactoryService))
```

Comment:
can you explain the old/new logic here.  (alternatively, you can try to keep indentation the same so that hte diff is clearer).

Reply:
there is no change in logic here, just 4 less spaces. Instead of
```
waitIndicator.Wait(
    title: EditorFeaturesResources.Automatic_Line_Ender,
    message: EditorFeaturesResources.Automatically_completing,
    allowCancel: false, action: w =>
    {
        ... code ...
    }
```
it's now
``` 
using (context.WaitContext.AddScope(allowCancellation: false, EditorFeaturesResources.Automatically_completing))
{
    ... code ...
}

Reply:
Btw, this looks sooo much cleaner in CodeFlow :)

---
In reply to: [159564396](https://github.com/dotnet/roslyn/pull/23769#discussion_r159564396) [](ancestors = 159564396)

Reply:
Not certain if i can use codeflow :-/